dvc/command/check_ignore.py

```diff
@@ -14,27 +15,71 @@ def __init__(self, args):
         self.ignore_filter = self.repo.tree.dvcignore
 
     def _show_results(self, result):
-        if result.match or self.args.non_matching:
-            if self.args.details:
-                logger.info("{}\t{}".format(result.patterns[-1], result.file))
-            else:
-                logger.info(result.file)
+        if not result.match and not self.args.non_matching:
+            return
 
-    def run(self):
-        if self.args.non_matching and not self.args.details:
-            raise DvcException("--non-matching is only valid with --details")
+        if self.args.details:
+            patterns = result.patterns
+            if not self.args.all:
+                patterns = patterns[-1:]
 
-        if self.args.quiet and self.args.details:
-            raise DvcException("cannot both --details and --quiet")
+            for pattern in patterns:
+                logger.info("{}\t{}".format(pattern, result.file))
+        else:
+            logger.info(result.file)
+
+    def _check_one_file(self, target):
+        result = self.ignore_filter.check_ignore(target)
+        self._show_results(result)
+        if result.match:
+            return 0
+        return 1
 
+    def _interactive_mode(self):
+        ret = 1
+        while True:
+            target = ask("")
+            if target == "":
+                logger.info(
+                    "Empty string is not a valid pathspec. Please use . "
+                    "instead if you meant to match all paths."
```

Comment:
What is a pathspec?
Can users actually send an empty string (e.g. `''`)? Or does this happen when you don't send any args? In the latter case the message could be simplified to something like "Please provide file or directory paths to check."

Reply:
@jorgeorpinel 
1. `pathspec` means the target to be checked
2. It is in an interactive mode where users can type string as the target, and this happened when users type send an empty string.

Reply:
> It is in an interactive mode
So maybe don't say anything and just go to the next `>` ? Seems intuitive enough that way. I just tried it and noticed it crashes after this error msg (i.e. it stops the interactive mode).

Reply:
p.s. when I send an empty string as target i.e. `dvc check-ignore ''` it just checks it (never matches even if `*` is a pattern in `.dvcignore`) and this error isn't presented. Seems slightly inconsistent. I vote to just remove it altogether 🙂

Reply:
Extracted to #4361